### ISSUE TITLE
Vaadin Client should automatically show a login form when access token is missing

### ISSUE THREAD

**Issue by vlukashov** [vaadin/vaadin-services#11]
When using the Vaadin Services JS client, I want it to automatically prompt for username and password if the user is not logged in and tries to make a call to the Vaadin Services backend, so that the default configuration of a Vaadin Serivces project automatically allows users to login without me writing any special code for that.

DoD:
 - Vaadin Client has a default login form UI (possibly, using [vaadin-login](https://github.com/vaadin/vaadin-login) if it is a free component)
 - when Vaadin Client does not have an access token and a backend call is initiated, it automatically displays a login from UI to collect the username and password, sends them to the authorization server to get an access token (and a refresh token as well), and then makes the original backend call with the received access token
 - when Vaadin Client gets an 'unauthorized' error in response to a call that included an access token, and when `refreshToken` is not a string, Vaadin Client displays the login form UI as in the previous point

The [common DoD](https://github.com/vaadin/vaadin-services/wiki/Definition-of-Done) applies.

**Comment by tmattsso** [comment]
I don't think any developer would want this as the default behaviour. Most apps have their own custom login screen, with custom logic for when login happens. How would this even work? a popup with a login form? a force redirect to login, destroying form data?

Please reconsider this feature. Personally, I think you should go with a callback route and let the dev control when and how the user inputs their credentials.

**Comment by manolo** [comment]
Good point. Yes callback is being implemented in https://github.com/vaadin/vaadin-connect/issues/55, does it covers the customisation you meant?

**Comment by tmattsso** [comment]
That ticket seems to cover the full case; my recommendation would be to drop this one.

**Comment by Legioth** [comment]
The idea here is that there should be a simple built-in login form that is used by default so that you can get started with the actual application logic immediately. It's then a separate issue that it should also be possible to define your own UI for logging in instead of using the default.

**Comment by Legioth** [comment]
Might make sense, even though it assumes projects are always created based on a starter that explicitly includes a login form.

The question then would only be what should happen if it's missing.

**Comment by vlukashov** [comment]
Proposal: if no login form is configured, the default would be to throw with a message like:
> https://vaad.in/errors/con-1234: please provide credentials in order to access secured methods with Vaadin Connect (tried to call `HelloService.hello()`)

That requires keeping a page with error descriptions somewhere on vaadin.com/docs and maintaining a list of short links on vaad.in
